src/redux/form/sagas/formAdd.saga.ts

```diff
@@ -40,14 +41,23 @@ export function* addFilmTask(data: {
     const filteredList = filterByGenre(sorting(allMovies, sortingType), genre);
     if (res) {
       yield put(filmListChanged(filteredList));
-    } else {
-      yield put(formAddFilm.failure("error"));
+      yield put(formAddFilm.success(film));
     }
   } catch (err) {
-    if (err instanceof Error) {
-      yield put(formAddFilm.failure(err.message));
-    } else {
-      yield put(formAddFilm.failure("Unknown error :("));
-    }
+    const errors: any = {};
+
+    err.forEach((el: string, index: number) => {
+      for (let key in FormFieldsName) {
+        if (el.includes(key)) {
+          const value = el.replace(
+            key,
+            FormFieldsName[key as keyof typeof FormFieldsName]
+          );
+          errors[key] = value;
+        }
+      }
+    });
+
+    yield put(formAddFilm.failure(errors));
```

Comment:
and what happens when there was 500 type error, you just missed it , and don't even write it into store.

Reply:
remember, that error handling is also crucial for good application